networks/suzuka/suzuka-client/src/load_soak_testing/mod.rs

```diff
@@ -0,0 +1,394 @@
+#[derive(Clone, Debug)]
+pub enum TestKind {
+	Load {
+		nb_scenarios: usize,
+	},
+	Soak {
+		min_scenarios: usize,
+		max_scenarios: usize,
+		duration: std::time::Duration,
+		nb_clycle: u32,
+	},
+}
+
+impl TestKind {
+	pub fn build_load_test(nb_scenarios: usize) -> Self {
+		TestKind::Load { nb_scenarios }
+	}
+	pub fn build_soak_test(
+		min_scenarios: usize,
+		max_scenarios: usize,
+		duration: std::time::Duration,
+		nb_clycle: u32,
+	) -> Self {
+		TestKind::Soak { min_scenarios, max_scenarios, duration, nb_clycle }
+	}
+}
+
+/// Execute the test scenarios define in the specified configuration.
+/// scenarios are executed by chunk. Chunk execution of scenario is done by a client.
+/// All clients are executed in a different thread in parallel.
+/// Clients execute scenario in a Tokio runtime concurrently.
+pub fn execute_test(config: ExecutionConfig, create_scenario: Arc<scenario::CreateScenarioFn>) {
+	tracing::info!("Start test scenario execution.");
+
+	let nb_scenarios = match config.kind {
+		TestKind::Load { nb_scenarios } => nb_scenarios,
+		TestKind::Soak { max_scenarios, .. } => max_scenarios,
+	};
+
+	//build chunk of ids. Start at 1. 0 mean in result execution fail before scenario can execute.
+	let ids: Vec<_> = (1..=nb_scenarios).collect();
+	let chunks: Vec<_> = ids
+		.into_iter()
+		.chunks(config.nb_scenario_per_client)
+		.into_iter()
+		.map(|chunk| {
+			(config.kind.clone(), chunk.into_iter().collect::<Vec<_>>(), create_scenario.clone())
+		})
+		.collect();
+	// Execute the client by id's chunk.
+	let exec_results: Vec<_> = chunks
+		.into_par_iter()
+		.map(|(kind, chunk, create_scenario)| {
+			let client = TestClient::new(chunk);
+			client.run_scenarios(kind.clone(), create_scenario.clone())
+		})
+		.collect();
+
+	let no_zero_exec_time: Vec<_> = exec_results
+		.into_iter()
+		.filter_map(|res| (res.average_execution_time_milli > 0).then_some(res))
+		.collect();
+
+	let average_exec_time = no_zero_exec_time
+		.iter()
+		.map(|res| res.average_execution_time_milli)
+		.sum::<u128>()
+		/ no_zero_exec_time.len() as u128;
+	let metrics_average_exec_time = serde_json::to_string(&average_exec_time)
+		.unwrap_or("Metric  execution result serialization error.".to_string());
+	tracing::info!(target:EXEC_LOG_FILTER, metrics_average_exec_time);
+	tracing::info!("Scenarios execution average_exec_time:{metrics_average_exec_time}");
+
+	tracing::info!("End test scenario execution.");
+}
+
+/// Run the specified scenarios concurrently using Tokio.
+#[derive(Default)]
+struct TestClient {
+	scenario_chunk: Vec<usize>,
+}
+
+impl TestClient {
+	fn new(scenario_chunk: Vec<usize>) -> Self {
+		TestClient { scenario_chunk }
+	}
+
+	fn run_scenarios(
+		self,
+		kind: TestKind,
+		create_scanario: Arc<scenario::CreateScenarioFn>,
+	) -> ClientExecResult {
+		// Start the Tokio runtime on the current thread
+		let rt = tokio::runtime::Builder::new_current_thread().enable_all().build().unwrap();
```

Comment:
Is the `unwrap()` intentional here? Do we want the whole process to crash here if the tokio runtime thread does not start? Or do we want to log and and capture an error and continue the whole process.